app/api/endpoints/lossClaim/list/v3/connector/ListLossClaimsConnector.scala

```diff
@@ -30,31 +33,67 @@ import scala.concurrent.{ ExecutionContext, Future }
 @Singleton
 class ListLossClaimsConnector @Inject() (val http: HttpClient, val appConfig: AppConfig) extends BaseDownstreamConnector {
 
-  def listLossClaims(request: ListLossClaimsRequest)(implicit
-      hc: HeaderCarrier,
-      ec: ExecutionContext,
-      correlationId: String): Future[DownstreamOutcome[ListLossClaimsResponse[ListLossClaimsItem]]] = {
+  type ListLossClaimsOutcome = DownstreamOutcome[ListLossClaimsResponse[ListLossClaimsItem]]
+
+  private val DEFAULT_TAX_YEARS: Seq[TaxYear] = List("2019-20", "2020-21", "2021-22", "2022-23").map(TaxYear.fromMtd)
+
+  private val NOT_FOUND_CODE = "NOT_FOUND"
+
+  def listLossClaims(
+      request: ListLossClaimsRequest)(implicit hc: HeaderCarrier, ec: ExecutionContext, correlationId: String): Future[ListLossClaimsOutcome] = {
+
     import request._
 
     val pathParameters = Map(
-      "taxYear"          -> taxYearClaimedFor.map(_.asDownstream),
       "incomeSourceId"   -> businessId,
       "incomeSourceType" -> typeOfLoss.flatMap(_.toIncomeSourceType).map(_.toString),
       "claimType"        -> typeOfClaim.map(_.toReliefClaimed.toString)
     ).collect { case (key, Some(value)) =>
       key -> value
     }
 
-    val (uri: DownstreamUri[ListLossClaimsResponse[ListLossClaimsItem]], downstreamSpecificParameters) = taxYearClaimedFor match {
-      case Some(taxYear) if taxYear.useTaxYearSpecificApi =>
-        (
-          TaxYearSpecificIfsUri[ListLossClaimsResponse[ListLossClaimsItem]](s"income-tax/claims-for-relief/${taxYear.asTysDownstream}/${nino.nino}"),
-          pathParameters.filterNot { case (key, _) => key == "taxYear" }
-        )
-      case _ => (IfsUri[ListLossClaimsResponse[ListLossClaimsItem]](s"income-tax/claims-for-relief/${nino.nino}"), pathParameters)
+    taxYearClaimedFor match {
+      case Some(taxYear) => makeRequestForTaxYear(taxYear, nino.nino, pathParameters)
+      case _             => makeRequestForDefaultTaxYears(nino.nino, pathParameters)
+    }
+  }
+
+  private def makeRequestForDefaultTaxYears(nino: String, params: Map[String, String])(implicit
+      hc: HeaderCarrier,
+      ec: ExecutionContext,
+      correlationId: String): Future[ListLossClaimsOutcome] = {
+
+    val requests = DEFAULT_TAX_YEARS.map(makeRequestForTaxYear(_, nino, params))
+
+    Future.sequence(requests).map(combineResponses)
```

Comment:
Nice :)

I'd suggest using Future.traverse though, as you then don't need the separate "requests" val. (traverse will still kick off all the Futures in parallel).

Reply:
Could also put `makeRequestForTaxYear` as a nested function, gets rid of a lot of the boilerplate. Can then shorten the whole method to something like:

```
    def send(taxYear: TaxYear): Future[ListLossClaimsOutcome] = {
      val uri = TaxYearSpecificIfsUri[ListLossClaimsResponse[ListLossClaimsItem]](s"income-tax/claims-for-relief/${taxYear.asTysDownstream}/$nino")
      get(uri, params.toSeq)
    }
    
    Future.traverse(DEFAULT_TAX_YEARS)(send).map(combineResponses)
```

Reply:
I didn't know about Future.traverse, that's useful to know! Just pushed a commit which uses that, and nests the two `makeRequest` functions.